src/Data/TypedByteArray.hs

```diff
@@ -22,3 +33,19 @@ fromList = TypedByteArray . byteArrayFromList
 {-# INLINE unsafeIndex #-}
 unsafeIndex :: Prim a => TypedByteArray a -> Int -> a
 unsafeIndex (TypedByteArray arr) = indexByteArray arr
+
+-- | Construct a 'TypedByteArray' of the given length by applying the function to each index in @[0..n-1]@.
+generate :: forall a. Prim a => Int -> (Int -> a) -> TypedByteArray a
+generate !n f = runST $ do
+    -- Allocate enough space for n correctly aligned elements
+    arr <- newByteArray $ alignment (undefined :: a) * n
```

Comment:
Couldn't aligment be too little? I.e. I could have 32 byte values that only need to be 8-byte aligned, for example. Or is `alignment x >= sizeOf x` guaranteed?

And if this follows C conventions, wouldn't `sizeOf` be suitably padded already to guarantee the right alignment?

Reply:
You are right, I got my terminology mixed up :+1: